### PR TITLE
Use libunwind from llvm-project submodule for musl targets

### PR DESCRIPTION
**Pull request by malbarbo**
This avoid downloading libunwind by using the scheme introduced in https://github.com/rust-lang/rust/pull/61544

### CODE REVIEW

**Comment by alexcrichton** [comment]
Nice! Definitely seems like a nice extension of what we have already. I think though this may need to keep the `#[link]` directive for the embedded `cfg` though, could the `cc` builder disable cargo metadata output and the `#[link]` directives would remain the same in the code?

**Comment by alexcrichton** [comment]
(it may also be worthwhile to enable temporarily for this PR the builders exercising musl tests to ensure those still work)

**Comment by malbarbo** [comment]
I tested it locally and was able to produce static and dynamic binaries. I also enabled test-various, which run musl tests.

**Comment by alexcrichton** [comment]
Looks great to me!

r=me with the PR cleaned up slightly

**Comment by malbarbo** [comment]
Done.

**Comment by nikic** [comment]
@bors r-

Failed in rollup #63345 on `dist-i586-gnu-i586-i686-musl`:

```
2019-08-07T09:31:26.2317594Z    Compiling libc v0.2.60
2019-08-07T09:31:26.2509078Z    Compiling compiler_builtins v0.1.18
2019-08-07T09:31:31.4941891Z    Compiling backtrace-sys v0.1.30
2019-08-07T09:31:33.3554962Z    Compiling unwind v0.0.0 (/checkout/src/libunwind)
2019-08-07T09:31:33.3555404Z error: failed to run custom build command for `unwind v0.0.0 (/checkout/src/libunwind)`
2019-08-07T09:31:33.3555915Z Caused by:
2019-08-07T09:31:33.3555915Z Caused by:
2019-08-07T09:31:33.3556485Z   process didn't exit successfully: `/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/release/build/unwind-a04f9435c6eb75e3/build-script-build` (exit code: 101)
2019-08-07T09:31:33.3556711Z --- stdout
2019-08-07T09:31:33.3556938Z cargo:rerun-if-changed=build.rs
2019-08-07T09:31:33.3557262Z cargo:rustc-link-search=native=/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/i686-unknown-linux-musl/release/build/unwind-3b3aa3b1105b96b5/out
2019-08-07T09:31:33.3558000Z running: "musl-g++" "-O2" "-ffunction-sections" "-fdata-sections" "-fPIC" "-m32" "-static" "-march=i686" "-Wl,-melf_i386" "-I" "../llvm-project/libunwind/include" "-std=c99" "-std=c++11" "-nostdinc++" "-fno-exceptions" "-fno-rtti" "-fstrict-aliasing" "-funwind-tables" "-o" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2-std/i686-unknown-linux-musl/release/build/unwind-3b3aa3b1105b96b5/out/../llvm-project/libunwind/src/Unwind-EHABI.o" "-c" "../llvm-project/libunwind/src/Unwind-EHABI.cpp"
2019-08-07T09:31:33.3558842Z --- stderr
2019-08-07T09:31:33.3559052Z thread 'main' panicked at '
2019-08-07T09:31:33.3559217Z 
2019-08-07T09:31:33.3559217Z 
2019-08-07T09:31:33.3559475Z Internal error occurred: Failed to find tool. Is `musl-g++` installed?
2019-08-07T09:31:33.3559806Z ', /cargo/registry/src/github.com-1ecc6299db9ec823/cc-1.0.35/src/lib.rs:2398:5
```

**Comment by malbarbo** [comment]
The musl-g++ compiler is missing in some builders, but it is not necessary. We can compile c++ code using musl-gcc because we do not link with libstdc++. (see https://github.com/rust-lang/rust/pull/63173/files#diff-3503e1a05300f03b6ff9df8257913e83R101)

**Comment by malbarbo** [comment]
Rebased.

File: src/libunwind/build.rs [hunk 1]
```diff
@@ -5,14 +5,14 @@ fn main() {
     let target = env::var("TARGET").expect("TARGET was not set");
 
     if cfg!(feature = "llvm-libunwind") &&
-        (target.contains("linux") ||
+        (target.contains("linux") && !target.contains("musl") ||
```
Comment (Keruspe) [inline src/libunwind/build.rs]: might want to add some parens wrt precendence between this `&&` and the `||` next to it?

Comment (malbarbo) [reply src/libunwind/build.rs]: Done.

Comment (malbarbo) [reply src/libunwind/build.rs]: Done

**Comment by Centril** [comment]
I think this failed in https://github.com/rust-lang/rust/pull/63607#issuecomment-521730607 but I'm not 100% sure. @bors r- meanwhile.

**Comment by Centril** [comment]
Seems like this PR wasn't to blame after all... @bors r=alexcrichton

**Pull request merged**